frontend/src/app/search/search-tabs.ts

```diff
@@ -0,0 +1,37 @@
+import { Params } from '@angular/router';
+import { StoreSync } from '../store/store-sync';
+import { Store } from '../store/types';
+import * as _ from 'lodash';
+
+export type SearchTab = 'search-results'|'visualizations'|'download';
+
+interface SearchTabState {
+    tab: SearchTab
+};
+
+export class SearchTabs extends StoreSync<SearchTabState> {
+    protected keysInStore = ['tab', 'visualize'];
+
+    constructor(store: Store) {
+        super(store);
+        this.connectToStore();
+    }
+
+    protected storeToState(params: Params): SearchTabState {
+        if (_.get(params, 'tab')) {
+            return { tab: params.tab }
+        } else if (_.get(params, 'visualize')) {
+            return { tab: 'visualizations' };
+        } else {
+            return { tab: 'search-results' }
+        };
+    }
+
+    protected stateToStore(state: SearchTabState): Params {
+        if (state.tab === 'search-results') {
+            return { tab: null };
+        } else {
+            return state;
+        }
```

Comment:
Familiarizing myself with the `store` stuff, which does look very neat. 
I wonder if something like a 'default state' would be beneficial to add as an option in `StoreSync`? In this case if the tab is `search-results`.

Reply:
Hm, good point. Filters have something like that, but perhaps it could be generalised to more models.